src/main/java/com/crowdin/cli/utils/PlaceholderUtil.java

```diff
@@ -0,0 +1,85 @@
+package com.crowdin.cli.utils;
+
+import com.crowdin.cli.utils.console.ConsoleUtils;
+import com.crowdin.common.models.Language;
+import org.apache.commons.io.FilenameUtils;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+public class PlaceholderUtil {
+
+    protected static final String PLACEHOLDER_ANDROID_CODE = "%android_code%";
+    protected static final String PLACEHOLDER_FILE_EXTENTION = "%file_extension%";
+    protected static final String PLACEHOLDER_FILE_NAME = "%file_name%";
+    protected static final String PLACEHOLDER_LANGUAGE = "%language%";
+    protected static final String PLACEHOLDER_LOCALE = "%locale%";
+    protected static final String PLACEHOLDER_LOCALE_WITH_UNDERSCORE = "%locale_with_underscore%";
+    protected static final String PLACEHOLDER_THREE_LETTERS_CODE = "%three_letters_code%";
+    protected static final String PLACEHOLDER_TWO_LETTERS_CODE = "%two_letters_code%";
+    protected static final String PLACEHOLDER_OSX_CODE = "%osx_code%";
+    protected static final String PLACEHOLDER_OSX_LOCALE = "%osx_locale%";
+    protected static final String PLACEHOLDER_ORIGINAL_FILE_NAME = "%original_file_name%";
+    protected static final String PLACEHOLDER_ORIGINAL_PATH = "%original_path%";
+
+    private List<Language> supportedLangs;
+    private List<Language> projectLangs;
+    private String basePath;
+
+    public PlaceholderUtil(List<Language> supportedLangs, List<Language> projectLangs, String basePath) {
+        if (supportedLangs == null || projectLangs == null || basePath == null)
+            throw new NullPointerException("in PlaceholderUtil.contructor");
+        for(Language projectLang : projectLangs) {
+            if (!supportedLangs.parallelStream().anyMatch(lang -> lang.getName().equals(projectLang.getName()))) {
```

Comment:
outer foreach can be replaced into something like this:
`projectLangs.stream()
  .findFirst(projectLang -> !supportedLangs.stream().anyMatch(lang -> lang.getName().equals(projectLang.getName())))
  .ifPresent(projectLang -> throw new RuntimeException("Proj contains langs that Crowdin doesn't support: " + projectLang.getName()))`;